Index: src/model.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/model.py b/src/model.py
--- a/src/model.py	(revision 3d96fac56380115aa946433ae9e86e893b1da3b0)
+++ b/src/model.py	(revision e4e0e018b48559b513c37662cd91cb104e0d3a8d)
@@ -90,7 +90,7 @@
                                                 mtf.range(params.mesh, tmp, dtype=tf.int32)),
                                        params.variable_dtype.activation_dtype)
                               ] if idx in params.masked_attention_dimensions else []),
-                          output_shape=qry.shape)
+                          output_shape=block_input.shape)
 
     lgt = mtf.einsum([qry, key], reduced_dims=[params.key_dim])
 
@@ -100,7 +100,7 @@
                         params.variable_dtype.activation_dtype) * -1e12
 
     lgt = mtf.exp(lgt - mtf.reduce_max(mtf.stop_gradient(lgt), reduced_dim=tmp))
-    return mtf.einsum([lgt, anonymize(val, dim)], qry.shape) / mtf.reduce_sum(lgt, reduced_dim=tmp)
+    return mtf.einsum([lgt, anonymize(val, dim)], block_input.shape) / mtf.reduce_sum(lgt, reduced_dim=tmp)
 
 
 def _rezero(params, block_input: mtf.Tensor, name_extras: typing.Tuple[str]) -> mtf.Tensor:
