Index: src/optimizers.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/optimizers.py b/src/optimizers.py
--- a/src/optimizers.py	(revision 24b9a7c2e4575f64399c1c4ec159e8853815573b)
+++ b/src/optimizers.py	(revision 5cdacd4729c59f2173bef5ebf276aeabadc31ab2)
@@ -102,7 +102,7 @@
                         if params.grad_accumulation > 1:
                             grad_buffer = get_variable(params, var, "grad_accumulation", var.shape)
                             step = mtf.cast(mtf.not_equal(mtf.mod(adam.global_step, params.grad_accumulation), 0),
-                                            var.dtype)
+                                            grad.dtype)
                             update_ops.append(mtf.assign(grad_buffer, grad + grad_buffer.value * (1 - step)))
                             grad = grad_buffer.value * step
                         weight_update, buffer = optim.apply_grad(grad, var)
